app/models/post.rb

```diff
@@ -1,3 +1,20 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
+
+  after_create :increment_post_counter
+  after_destroy :decrement_post_counter
```

Comment:
[OPTIONAL] Using `after_save` would have been more straightforward and saved you the stress of creating two methods to increment and decrement. But good job using callbacks.

Reply:
Thanks! will look for that in the future!